samples/bluetooth/bap_broadcast_sink/src/main.c

```diff
@@ -1023,6 +1064,97 @@ static int pa_sync_create(void)
 
 	return bt_le_per_adv_sync_create(&create_params, &pa_sync);
 }
+#if defined(CONFIG_TARGET_BROADCAST_CHANNEL)
+static uint32_t keep_least_significant_allocation_match(
+	uint32_t bitfield, enum bt_audio_location audio_allocation[BT_ISO_BIS_INDEX_MAX],
+	uint32_t target_audio_alloc)
+{
+	enum bt_audio_location audio_alloc = BT_AUDIO_LOCATION_MONO_AUDIO;
+	uint32_t result = 0U;
+
+	for (uint8_t i = 0; i < BT_ISO_BIS_INDEX_MAX && bitfield != 0; i++) {
+		uint32_t lsb = bitfield & -bitfield; /* extract lsb */
+
+		if (lsb) {
```

Comment:
```suggestion
		if (lsb != 0) {
```

Coding guidelines state that all control statements' expression shall have a essentially Boolean type (rule 85 of https://docs.zephyrproject.org/latest/contribute/coding_guidelines/index.html)